ITISFoundation/osparc-simcore
# is288/node filters

## What do these changes do?


* This adds the possibility to create filters that can be used by UI elements to change their visibility or to trigger some other visuals.
* Corrects small detected issues.

## Related issue number


connects to #670 

## Test it
* Review the code and check that everything makes sense (use of Qooxdoo's own implementation of interfaces, mixins, inheritance...).
* Try the text filtering for the nodes in the workbench and see that it doesn't bug.
* Try navigating away from the workbench to different view and go back and check that everything works.
* Check the visuals. Would you highlight/hide the nodes in some other way?

## Screenshots
![filtering2](https://user-images.githubusercontent.com/4764217/56351931-7a07ce80-61ce-11e9-89e5-e66956fe259d.gif)


## Checklist


[waffle.io]:https://waffle.io/marketing-assets/documents/waffleio_cheatsheet_v1.pdf?utm_source=blog&utm_medium=cheatsheet-ctabutton&utm_campaign=cheatsheet
